### DISCUSSION TITLE
Read sourceDir during config stage

### DISCUSSION THREAD

**Discussion by Hecatron** [vuepress/core#650] [Q&A] [answered]
Hi, does anyone know how to read the ${sourceDir} during the configuration stage?

I'm basically trying to setup my own implementation of vuepress-bar that generates side / top navigation menus automatically in code
but not as a plugin, instead the code runs / returns the needed input for the config
Something that will work between different themes since [vuepress-theme-mix](https://vuepress-theme-mix.vercel.app/guide/sidebar.html) for example uses it's own implementation.

Since this isn't a plugin but runs during the configuration stage, I'm wondering on the best way to do this.

**Comment by Mister-Hope** [comment, marked as answer]
You can try to invoke it in vuepress config file.

```ts
import { defineUserConfig} from '@vuepress/cli';
import { helper } from 'your-plugin'

const config = defineUserConfig({
  // ...
});

// do somthing with helper using the config
helper(config)

return config;
```

And I am assuming you know how to infer source Dir from args and pwd.
